atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -89,17 +106,33 @@ public void create(byte[] tableMetadata) {
 
     @Override
     public void drop() {
-        executeIgnoringError("DROP TABLE " + prefixedTableName() + " PURGE", "ORA-00942");
-        executeIgnoringError("DROP TABLE " + prefixedOverflowTableName() + " PURGE", "ORA-00942");
+        executeIgnoringError("DROP TABLE " + getInternalTableName(prefixedTableName()) + " PURGE", "ORA-00942");
+        conns.get().executeUnregisteredQuery(
+                "DELETE FROM " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE
+                        + " WHERE table_name = ?", prefixedTableName());
+
+        TableSize tableSize = TableSizeCache.getTableSize(conns, tableRef, config.metadataTable());
+        if (tableSize.equals(TableSize.OVERFLOW)) {
+            executeIgnoringError("DROP TABLE " + getInternalTableName(prefixedOverflowTableName()) + " PURGE",
+                    "ORA-00942");
+            conns.get().executeUnregisteredQuery(
+                    "DELETE FROM " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE
+                            + " WHERE table_name = ?", prefixedOverflowTableName());
+        }
+
         conns.get().executeUnregisteredQuery(
                 "DELETE FROM " + config.metadataTable().getQualifiedName()
                 + " WHERE table_name = ?", tableRef.getQualifiedName());
     }
 
+
     @Override
     public void truncate() {
-        executeIgnoringError("TRUNCATE TABLE " + prefixedTableName(), "ORA-00942");
-        executeIgnoringError("TRUNCATE TABLE " + prefixedOverflowTableName(), "ORA-00942");
+        executeIgnoringError("TRUNCATE TABLE " + getInternalTableName(prefixedTableName()), "ORA-00942");
+        TableSize tableSize = TableSizeCache.getTableSize(conns, tableRef, config.metadataTable());
+        if (tableSize.equals(TableSize.OVERFLOW)) {
+            executeIgnoringError("TRUNCATE TABLE " + getInternalTableName(prefixedOverflowTableName()), "ORA-00942");
```

Comment:
The ability to inject table names into sql strings exist all over the code base.  For our current impl we are operating on the assumption that tables are never user supplied (as is currently true from my understanding).  Might be best to discuss off thread.